src/OmniSharp.DotNetTest/Helpers/TestFeaturesDiscover.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using OmniSharp.Abstractions.Services;
+
+namespace OmniSharp.DotNetTest.Helpers
+{
+    [Export(typeof(ISyntaxFeaturesDiscover))]
+    public class TestFeaturesDiscover : ISyntaxFeaturesDiscover
+    {
+        public bool NeedSemanticModel { get; } = true;
+
+        public IEnumerable<string> Discover(SyntaxNode node, SemanticModel semanticModel)
+        {
+            if (node is MethodDeclarationSyntax)
+            {
+                var method = node as MethodDeclarationSyntax;
+                if (IsTestMethod(method, semanticModel))
+                {
+                    var methodName = semanticModel.GetDeclaredSymbol(node).ToDisplayString();
+                    methodName = methodName.Substring(0, methodName.IndexOf('('));
+
+                    yield return $"XunitTestMethod:{methodName}";
```

Comment:
That's why the endpoint is named `rundotnettest` to eliminate the ambiguity. Moving forward, we can:
1. add new endpoint for other types of test
2. make run test endpoint general to support extensible test provider.

but for now, I'd like to keep the scope narrow.